tools/test/examples/examples_cmake.json

```diff
@@ -156,7 +156,7 @@
               "NFC_SmartPoster"
           ],
           "features" : [],
-          "targets" : ["NUCLEO_F401RE"],
+          "targets" : ["NUCLEO_F401RE", "DISCO_L475VG_IOT01A"],
```

Comment:
We can't/shouldn't add DISCO_L475VG_IOT01A, because only NUCLEO_F401RE supports both NFC examples. The compilation failure with Mbed CLI 2 is expected behaviour, whereas Mbed CLI 1 lacks a compatibility check and always builds the NFC driver regardless, so it passes unexpectedly. See #14527.